clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1143,10 +1144,10 @@ public SelectorMetrics(Metrics metrics, String metricGrpPrefix, Map<String, Stri
 
         private Meter createMeter(Metrics metrics, String groupName, Map<String, String> metricTags,
                 SampledStat stat, String baseName, String descriptiveName) {
-            MetricName rateMetricName = metrics.metricName(baseName + "-rate", groupName,
-                            String.format("The number of %s per second", descriptiveName), metricTags);
-            MetricName totalMetricName = metrics.metricName(baseName + "-total", groupName,
-                            String.format("The total number of %s", descriptiveName), metricTags);
+            MetricName rateMetricName = metrics.metricName((baseName + "-rate").intern(), groupName,
```

Comment:
i'd add a comment explaining the intern()s - just a one-liner like "these string are very repetitive and may have 10Ks+ of identical objects in memory,. they are also ~low QPS. so we intern() them"

Reply:
Makes sense!

Reply:
Does intern() finally work without causing OOM?

Reply:
I think the newer java versions 7/8/9 have some improvements to `intern()` that should reduce the probability of an OOM. We are also using intern() on a few strings that we know for sure are repetitive. I think we should be fine.